Fix/e2e fe tests

ralucaStan opened the pull request:
Fix checkout test 
- make tests independent & test also for empty cart case
- make sure Order summary is expanded with better selector

Remove unnecessary localStorage operations in tests.

This PR also touches some jest and puppeteer settings:
- increase puppeteer timeout on CI to prevent timeouts
- add ts to moduleFileExtensions in jest, to allow imports from ts files

Most importantly it removes the interception of console.log in tests with the method `observeConsoleLogging()`.
Reasons for removing:
-  pollutes our test logs with runtime console warnings
-  prevents console.log from inside tests
- it's mostly legacy code, as big part of the logic was done for Puppeteer 1.6.1, but since 3.0.0, `message.text()` returns string so the workaround is not necessary.

ralucaStan committed:
comment out failing php test

mikejolley committed:
Revert "comment out failing php test"

This reverts commit 7c40e8caf3aa32e35e3b70eb32051251b06e0613.

mikejolley committed:
Fix USD from merge conflict

ralucaStan committed:
test to see checkout page title is correct

ralucaStan committed:
unde jest config change

ralucaStan committed:
Fix checkout test

- make tests independent & test also for empty cart case
- make sure Order summary is expanded with better selector

ralucaStan committed:
Remove unnecessary localStorage operations from tests

senadir commented on tests/e2e/specs/frontend/checkout.test.js:
```diff
@@ -36,12 +36,7 @@ describe( `${ block.name } Block (frontend)`, () => {
-				'wc-blocks_dismissed_compatibility_notices'
-			);
-		} );
+		// prevent CartCheckoutCompatibilityNotice from appearing
```
Unrelated but is this needed here? CartCheckoutCompatibilityNotice shouldn't appear in checkout to begin with.

ralucaStan replied:
goToCheckoutBlock method goes to the editor, on the Checkout block page to get the permalink in order to navigate to that url

senadir reviewed (approved):
This looks good so far. Let's see if tests pass.

ralucaStan committed:
Adjust navigation to page in test

ralucaStan committed:
test checkout block page on CI

ralucaStan committed:
test checkout page on ci

ralucaStan committed:
Test Empty cart step on CI

ralucaStan committed:
Use emptyCart method instead of removing individual products

ralucaStan committed:
reset browser before each frontend 2e2 assertion

cart is not empty when starting new tests

ralucaStan committed:
go to cart page before removing item from cart

ralucaStan committed:
Clean cart and checkout tests

ralucaStan committed:
Remove logging observing before tests

This removes unnecessary console.log interception. Big part of the logic was done for Puppeteer 1.6.1, but since 3.0.0 message.text() returns string

ralucaStan commented:
@senadir please take another look. 
- I've fixed the failing test,  basically cart was not being emptied before the checkout test, so the empty cart state was never displayed
- I've also added some jest and puppeteer settings with things I've noticed during debugging. Please see the PR description again

senadir commented on tests/e2e/config/jest.setup.js:
```diff
@@ -1,18 +1,16 @@
 
 // Set the default test timeout.
-let jestTimeoutInMilliSeconds = 40000;
+let jestTimeoutInMilliSeconds = 60000;
```
Waiting 60 seconds for a timeout seems excessive, anything that is logical and takes more than 40 seconds?

ralucaStan replied:
I actually wanted to make it bigger & make sure that if the timeout expires then the element isn't really found on the page. This setting would eliminate the slow UI as a reason and highlight that there is an issue with the page. As this only affects failing tests I don't think the impact is that big. We could try it out and see the outcome.

senadir commented on tests/e2e/config/jest.setup.js:
```diff
@@ -108,104 +102,12 @@ function removePageEvents() {
- * Adds a page event handler to emit uncaught exception to process if one of
- * the observed console logging types is encountered.
- */
-function observeConsoleLogging() {
```
Now that you deleted this, it means we won't catch console errors right? is this something that we want? couldn't we just skip `warning` type?

ralucaStan replied:
@senadir You make a good point here. I've allowed console.error to surface, but only for debugging reasons. It's good to see them if they appear in the logs. The old comment mentioned that console.error breaks the tests, but that is no longer the case.

senadir reviewed (approved):
So far this looks good to me.

I get the value of cleaning console warning but we would be missing errors emitted by our own blocks unless I'm wrong?

ralucaStan committed:
Allow console.error messages to surface in our tests.

Although they don't cause the test to fail it might be a good addition for debugging purposes.

ralucaStan commented:
> So far this looks good to me.
> 
> I get the value of cleaning console warning but we would be missing errors emitted by our own blocks unless I'm wrong?

@senadir You make a good point here. I've allowed `console.error` to surface, but only for debugging reasons. It's good to see them if they appear in the logs. The old comment mentioned that `console.error` breaks the tests, but that is no longer the case.

senadir commented:
Ready to ship!
